Update DataSure documentation (#171)
PovertyAction/ipa-research-data-science-hub: data-quality/datasure/how-to-datasure.qmd

## Key Takeaways

- DataSure follows a seven-step workflow: create a project, import data, prepare data, configure checks, review DQA reports, correct data, and export a replication package.
- First-time users should start with Demo Mode, which provides a guided walkthrough using sample household survey data.
- Data can be imported directly from a SurveyCTO server or uploaded as a CSV, Excel, Stata, or JSON file.
- Quality check reports update automatically each time you import new data.

> - The Replication Package exports auto-generated Stata do-files and audit logs, allowing anyone to reproduce your corrected dataset from raw source data.

Reviewer: Fix this Vale error. 
```suggestion
- The Replication Package exports autogenerated Stata do-files and audit logs, allowing anyone to reproduce your corrected dataset from raw source data.
```

DataSure opens in your default web browser, typically at `http://localhost:8501`. Keep the terminal window open while you work; closing it will stop the application.

You will land on the **Start Here** page, which is your home base for creating, selecting, and managing projects.

> ![The DataSure Start Here page showing the welcome header, Learn More expandable section, and project selector.](images/datasure_start_page.png){fig-alt="DataSure Start Here page with welcome header, a collapsible Learn More section, and the project selector dropdown." .lightbox}

Reviewer 2: Ishmail, the cover image changed. Please see images/datasure_newcover.png
![datasure_newcover.png](https://github.com/user-attachments/assets/dc4853bd-f61f-4fc9-8e5a-3206220dad38)

## Getting Familiar: Try Demo Mode First

If this is your first time using DataSure, start with Demo Mode before working with your own data. Demo Mode loads a pre-built sample project and guides you through all seven steps of the DataSure workflow with realistic household survey data. It is the fastest way to understand how each part of the tool works without any risk to real project data.

### Starting Demo Mode

1. On the **Start Here** page, select **DataSure Demo** from the project dropdown.
2. If you have not run the demo before, click **Start Demo**. DataSure will automatically load the sample datasets.
3. If you have run the demo previously, you will see two options:
   - **Resume Demo**: Continue from where you left off, including any prep steps, configurations, and corrections you made.
   - **Restart Demo**: Delete all demo progress and start fresh. DataSure will ask you to confirm before proceeding, since this cannot be undone.

> > *[GIF or short video: Selecting DataSure Demo from the dropdown and clicking Start Demo, showing the automatic data loading and navigation to the Import Data page.]*

Reviewer: ❓ Did you intend to add the video or GIF?

Reviewer 2: I am suggesting a video

Reviewer 2: https://github.com/user-attachments/assets/9bdcd252-5f3c-4f5b-a0ff-931412f6d46b

Author: Thanks for the vid. There is an updated start page and I have added an image to show that.

| Quotient | Add a column with the row-wise quotient of two columns | Divide a total score by maximum possible to compute a proportion |
| Index | Add a column containing the row number (0-based integer index) | Add a row number for reference when there is no existing unique identifier |
| UUID | Add a column of randomly generated universally unique identifiers | Generate a key column when the dataset has no unique row identifier |
| Random | Add a column of random numbers drawn from a uniform distribution | Generate a random assignment or sampling variable |

---

#### Remove Column

Use **Remove Column** to delete one or more columns from the dataset. Select the column or columns to remove from the list.

::: {.callout-warning}
Removing a column is a preparation step that is applied before checks run. Removing a column that you later need for check configuration will require you to delete the step and re-add it.
:::

---

#### Remove Row

Use **Remove Row** to filter out rows that match a condition. Select a column, a condition, and any required values.

**Conditions**

| Condition | What it Does | Example |
| --------- | ------------ | ------- |

> | Missing | Keep only rows where the column value is null or empty | Remove rows with no submission date |

Reviewer: ```suggestion
| Missing | Keep only rows where the column value is null or empty | Remove rows where submission date is empty |
```

> | Not Missing | Keep only rows where the column value is not null | Remove rows missing the respondent ID |

Reviewer: ```suggestion
| Not Missing | Keep only rows where the column value is not null | Remove rows where respondent ID is missing |
```

## Step 4: Configure Quality Checks

The **Configure Checks** page is where you tell DataSure which dataset to analyze and how the columns in your dataset map to the identifiers DataSure needs to run its checks. Each configuration you create becomes a separate **DQA Report** page in the sidebar.

Navigate to Configure Checks from the sidebar.

### Creating a check configuration

1. Click **Add New Check Configuration**.
2. Enter a **Page Name** for this configuration. This becomes the label for the DQA Report page in the sidebar. Keep it descriptive but concise, for example: `Household HFCs` or `Midline Survey Checks`. The name must be between 1 and 20 characters.
3. Select your **Survey Dataset** from the dropdown. This should be the main survey dataset you imported.
4. Configure the column fields as described in the sections below.
5. Click **Submit** to create the configuration.

DataSure validates your inputs when you submit. If any required fields are missing or column selections are invalid, it will show a specific error message.

> ![The Add New Check Configuration dialog fully filled out with example values for all fields.](images/datasure_config_oage_prefilled.png){fig-alt="DataSure Configure Checks page showing the Add New Check Configuration dialog with all fields completed, including page name, survey dataset, key column, ID column, enumerator column, date column, and other optional settings." .lightbox}

Reviewer: ⭐ ⭐ ⭐ fix the file name and path
```suggestion
![The Add New Check Configuration dialog fully filled out with example values for all fields.](images/datasure_config_page_prefilled.png){fig-alt="DataSure Configure Checks page showing the Add New Check Configuration dialog with all fields completed, including page name, survey dataset, key column, ID column, enumerator column, date column, and other optional settings." .lightbox}
```

Navigate to **Replication Package** from the sidebar.

### What the replication package contains

The zip file is organized into a structured folder layout:

```
replication_[project]_[page]/
├── 0_README.txt
├── 1_docs/
│   ├── 1_surveys/          ← SurveyCTO questionnaire (.xlsx), if applicable
│   ├── 2_codebooks/
│   └── 3_notes/
├── 2_scripts/
│   ├── 0_main.do           ← Master script that runs all others in order
│   ├── 1_install_packages.do
│   ├── 2_import_data.do    ← Imports the raw CSV into Stata
│   ├── 3_prepare_data.do   ← Applies all prep steps from the Prepare Data log
│   └── 4_corrections.do    ← Applies all corrections from the Correct Data log
├── 3_data/
│   ├── 1_raw/              ← Raw survey data as CSV
│   ├── 2_intermediate/     ← Generated when scripts are run
│   └── 3_final/            ← Generated when scripts are run
└── 4_output/
    ├── 1_tables/
    ├── 2_figures/
    └── 3_logs/
        ├── correction_log.csv
        └── prep_log.csv
```

> The Stata do-files (`2_import_data.do`, `3_prepare_data.do`, `4_corrections.do`) are **auto-generated** by DataSure from the preparation steps and corrections you logged during your project. You do not need to write any Stata code yourself. Running `0_main.do` in Stata will reproduce the full corrected dataset from scratch.

Reviewer: Fix Vale error

```suggestion
The Stata do-files (`2_import_data.do`, `3_prepare_data.do`, `4_corrections.do`) are **autogenerated** by DataSure from the preparation steps and corrections you logged during your project. You do not need to write any Stata code yourself. Running `0_main.do` in Stata will reproduce the full corrected dataset from scratch.
```
